cou/cli.py

```diff
@@ -108,78 +113,70 @@ async def continue_upgrade() -> bool:
     return False
 
 
-async def analyze_and_plan(args: CLIargs) -> UpgradePlan:
-    """Analyze cloud and generate the upgrade plan with steps.
+async def get_model(args: CLIargs) -> Model:
+    """Connect and get the juju model.
 
     :param args: CLI arguments
     :type args: CLIargs
-    :return: The generated upgrade plan.
-    :rtype: UpgradePlan
+    :return: Juju Model
+    :rtype: Model
     """
     model = Model(args.model_name)
     progress_indicator.start(f"Connecting to '{model.name}' model...")
     await model.connect()
     logger.info("Using model: %s", model.name)
     progress_indicator.succeed(f"Connected to '{model.name}'")
+    return model
 
+
+async def analyze_and_generate_plan(model: Model, args: CLIargs) -> UpgradePlan:
+    """Analyze the cloud and generate plan for cloud upgrade.
+
+    :param model: The model to run on
+    :type model: Model
+    :param args: CLI arguments
+    :type args: CLIargs
+    :return: The generated upgrade plan.
+    :rtype: UpgradePlan
+    :raises CloudVerificationError: when cloud is not ready for upgrade
+    """
     progress_indicator.start("Analyzing cloud...")
     analysis_result = await Analysis.create(model, skip_apps=args.skip_apps)
     logger.info(analysis_result)
     progress_indicator.succeed()
 
-    progress_indicator.start("Generating upgrade plan...")
-    upgrade_plan = await generate_plan(analysis_result, args)
+    progress_indicator.start("Verifying cloud...")
+    await verify_cloud(analysis_result, args=args)
     progress_indicator.succeed()
 
-    return upgrade_plan
-
+    if errors := PlanStatus.error_messages:
+        for error in errors:
+            logger.error(error)
+        raise CloudVerificationError(
+            "Running upgrades will not be possible until problems indicated in the errors "
+            "are resolved"
+        )
```

Comment:
I think this is a kind of regression. ManSol complained about not being able to generate and see upgrade plan if we detect some issue.

I think we should block the upgrade, but not the plan generation.

Reply:
I think that's a fair request

Reply:
Done